### PR TITLE
W-11644077:Implement useDefaultEndpoint in MMP &  W-11644077:The code of the MMP is missing the Resources property

### PR DESCRIPTION
**Pull request by egrave**
W-11644077: The new property is not called useDefaultEndpoint as expected but generateDefaultPublicUrl. It was already working in CH2 and now it's added to RTF.
W-11644077: the problem was that the settings were instantiated as ch2deploymentsetting for rtf  deployments.

### CODE REVIEW

File: mule-deployer/src/main/java/org/mule/tools/deployment/cloudhub2/RequestBuilderCh2.java [hunk 1]
```diff
@@ -57,6 +59,25 @@ protected ApplicationModify buildApplicationModify() {
     return applicationModify;
   }
 
+  protected RuntimeFabricDeploymentSettings resolveDeploymentSettings(RuntimeFabricDeploymentSettings settings)
+      throws DeploymentException {
+    RuntimeFabricDeploymentSettings resolvedDeploymentSettings =
+        new Cloudhub2DeploymentSettings((Cloudhub2DeploymentSettings) settings);
+    String targetId = super.resolveTargetId();
+    String muleVersion = deployment.getMuleVersion().get();
+    String tag = resolveTag(targetId, muleVersion);
```
Comment (rodriguez-jorge) [inline mule-deployer/src/main/java/org/mule/tools/deployment/cloudhub2/RequestBuilderCh2.java:68]: Maybe:
```   
    if (tag == null) {
        throw new DeploymentException("Could not resolve tag for this mule version");
    }
    
    resolvedDeploymentSettings.setRuntimeVersion(muleVersion + ":" + tag);
```

is more readable.

**Review by rodriguez-jorge — APPROVED** [review]
LGTM

**Review by rodriguez-jorge — APPROVED** [review]
LGTM

**Review by mpereyra-salesforce — APPROVED** [review]
LGTM!!!

**Pull request merged**